hazelcast/src/main/java/com/hazelcast/config/InstanceTrackingConfig.java

```diff
@@ -53,6 +57,16 @@
 
     public InstanceTrackingConfig() {
         super();
+        try {
+            Class<?> helper = Class.forName("com.hazelcast.license.util.LicenseHelper");
+            Method getLicense = helper.getMethod("getBuiltInLicense");
+            // enabled for NLC build
+            enabled = getLicense.invoke(null) != null;
```

Comment:
I would add a log message to let the user know that we forcibly enabled the instance tracking feature, because it's an NLC build. I think the behavior can be a bit confusing, since by default we have this feature disabled and the config schemas declare `enabled` as `false` by default. Alternatively, we can generate a different default config for the NLC builds, but I guess it needs a bit more work.

Reply:
I have to figure out some other way instead of logging it here. A user might have called `new InstanceTrackingConfig().setEnabled(false)` and if I log here, I will log something like "auto-enabling instance tracking" as soon as the config class is constructed, even though it's explicitly disabled in the same line.

I don't understand the approach with different default config. I'll try to mark somehow if `setEnabled()` was called. If it wasn't called, I'll assume it was auto-enabled and print the log when starting the EE instance.

Reply:
Or, I can improve the documentation. Let's try that first.

Reply:
Made some improvements, please take a look.